devito/passes/clusters/cse.py

```diff
@@ -270,8 +274,12 @@ def _toposort(exprs):
 
     def choose_element(queue, scheduled):
         # Try to honor temporary names as much as possible
-        first = sorted(queue, key=lambda i: str(i.lhs)).pop(0)
-        queue.remove(first)
+        tmps = [i for i in queue if isinstance(i.lhs, CTemp)]
```

Comment:
What is this change doing?

Reply:
Prevent re-ordering equations that are not CSE related. We only build the DAG edges for CTemp, so we can only consider those here for sorting by name, as the dependency edges will be missing for all the other equations, and sorting them by name might lead to wrong ordering.

This toposort is only meant to sort the CSE temps and nothing else.